### ISSUE TITLE
Detach the reprojection loss mask needed?

### ISSUE THREAD

**Issue by DaoyiG** [nianticlabs/manydepth#7]
Hi,

Thanks a lot for this open repo and your interesting paper! I have a question wrt a detail in your code. In ```trainer.py``` line 607
```
reprojection_loss = reprojection_loss * reprojection_loss_mask 
reprojection_loss = reprojection_loss.sum() / (reprojection_loss_mask.sum() + 1e-7) 
```
I wonder does this```reprojection_loss_mask``` need to be detached?

Thanks!

**Comment by mdfirman** [comment]
Thanks for your interest!
As far as I can tell, `reprojection_loss_mask` doesn't need to be detached as it is already created from mathematical operations which cannot be backpropogated through:

```
            # we are using automasking
            all_losses = torch.cat([reprojection_loss, identity_reprojection_loss], dim=1)
            idxs = torch.argmin(all_losses, dim=1, keepdim=True)
            reprojection_loss_mask = (idxs == 0).float()
```
As you can see in the final line here, the mask is created via an equality comparison (`idxs == 0`), through which no gradient can flow. So it effectively 'already detached'. 

If the mask was computed in a different way then yes it might need to be detached.

Hope that makes sense.

Thanks

Michael
